astropy/io/fits/scripts/fitsdiff.py

```diff
@@ -232,7 +232,16 @@ def match_files(paths):
                 sys.exit(2)
             filelists.append(files)
         elif os.path.isdir(path):
-            filelists.append([os.path.abspath(f) for f in os.listdir(path)])
+            if not filelists:
+                filelists.append([os.path.abspath(os.path.join(path, f)) for f in os.listdir(path)])
```

Comment:
The idea is fine but the existing code was already confusing, and imo testing directly `if not filelists` is not clear enough. Not sure what the best option is, maybe using `for i, path in enumerate(paths):` and testing for `i == 0` ? Also if can add some comments to explain the reasoning.

Reply:
Or we can define a more verbal counter!

Reply:
Or I can just a leave a comment beside it? U say!